### ISSUE TITLE
Memory Management

### ISSUE THREAD

**Issue by codectile** [codectile/cimulator#8]
There's a lot of issue regarding memory management, maybe I have to re-code everything. Looking forward to solve the issues before I go inactive.

**Comment by Crayder** [comment]
"go inactive?"

**Comment by codectile** [comment]
Yes, I will take long break for 4 months. I have got something to focus on. On topic: bullet follows a policy, ***whoever allocates the memory deletes its as well*** and I did not follow that policy too well. Memory management is one of the hardest issue in every program dealing with memory creation and deletion.

**Comment by codectile** [comment]
@Crayder I don't script much and I guess you script a lot. So it would be better if I ask you that, "will scripters prefer using environment based physics only?"
I recently came to know that creating an API allows modification of objects at run-time does not come out well and it links to a potential crash. The memory management issues bounds cimulator to only simulate environment based physics with some scripting precautions. If yes, I'll look forward to rebuild this project, else I'll abort this project.

**Comment by Crayder** [comment]
I'm not quite sure what you're asking... Like, what do you mean by "environment based"?

**Comment by codectile** [comment]
like blowing up objects, user created monument destruction etc...

**Comment by Crayder** [comment]
I'm still confused. Like whether or not to use bullet's built in physics for things like that? If so, then yes of course.

**Comment by codectile** [comment]
That means you want to use, right?
I'll make things much more clear after your reply.

**Comment by Crayder** [comment]
Yes I'd like to use those features, including all of the settings that have been added (friction, gravity, mass, restitution... and any more).
